### PR TITLE
cmd/sys-mgmt-executor/main.go: use json output, misc cleanups

### LINKED ISSUE #1129: Streamline the parsing of the output from the `docker` command being serviced by system-management.
- Based on excellent PR feedback from @anonymouse64, the goal of this task will be to streamline the parsing of the output from the `docker` command.
- The details below are from the feedback from @anonymouse64 which appear in [PR-1122](https://github.com/edgexfoundry/edgex-go/pull/1122) :

> The `docker` command has a flag to make the output conform to JSON. While an easy way to parse that JSON would be to import the docker libraries where that JSON schema is defined in go, that's a heavy handed way and instead it's fine to simply just define the fields you care about from the JSON inside a struct with proper structures, and then when parsing specify to ignore unknown fields so the parsing doesn't fail. For example see:

```
$ docker ps -a --format '{{json .}}' | jq
{
  "Command": "\"date\"",
  "CreatedAt": "2019-02-22 08:49:12 +0100 CET",
  "ID": "e477204cc97d",
  "Image": "ubuntu",
  "Labels": "",
  "LocalVolumes": "0",
  "Mounts": "",
  "Names": "clever_edison",
  "Networks": "bridge",
  "Ports": "",
  "RunningFor": "6 seconds ago",
  "Size": "0B",
  "Status": "Created"
}
```

> Also note that docker outputs it's JSON as a stream, which means that independent JSON objects are output immediately after one another, rather than being put in a list. See https://golang.org/pkg/encoding/json/#example_Decoder_Decode_stream for an example of how to handle this with the standard JSON library.

### PR DESCRIPTION
**Pull request by anonymouse64**
* Remove newlines at start and end of usage message
* Remove unnecessary variables when running docker commands
* Remove duplicated logging messages and instead wrap errors and return
* Return error as second return value when checking status of container
  after an operation
* Fix error where start operation result wasn't being checked because
  fallthrough wasn't being used
* Get status of docker container directly using json formatted docker
  inspect command and decoding the output

Fixes #1129 

To test, I used the docker-compose file from developer-scripts for delhi: https://github.com/edgexfoundry/developer-scripts/blob/master/compose-files/docker-compose-delhi-0.7.1.yml.

The following things should be tested:
- [ ]  make sure that starting and stopping a real service works
    - `docker-compose -f docker-compose-delhi-0.7.1.yml up -d`
    - `sys-mgmt-executor stop edgex-support-notifications`
    - `docker inspect edgex-support-notifications | jq '.[].State | .Status'` →shows `"exited"`
    - `sys-mgmt-executor start edgex-support-notifications`
    - `docker inspect edgex-support-notifications | jq '.[].State | .Status'` → shows `"running"`
- [ ]  make sure that starting/stopping a fake service fails
    - `docker-compose -f docker-compose-delhi-0.7.1.yml up -d`
    - `./sys-mgmt-executor stop not-a-svc` → graceful error
- [ ]  make sure that running with invalid operation fails gracefully
    - `docker-compose -f docker-compose-delhi-0.7.1.yml up -d`
    - `./sys-mgmt-executor do-other-thing edgex-support-notifications` → produces graceful error
    - `docker inspect edgex-support-notifications | jq '.[].State | .Status'` → unchanged
- [ ]  make sure that starting/stopping with errors fails gracefully (do this by removing the container so it can't be referenced by name to start/stop)
    - `docker-compose -f docker-compose-delhi-0.7.1.yml up -d`
    - `docker stop edgex-support-notifications && docker rm edgex-support-notifications`
    - `./sys-mgmt-executor stop edgex-support-notifications` → produces graceful error
    - `./sys-mgmt-executor start edgex-support-notifications` → produces graceful error

### CODE REVIEW

File: cmd/sys-mgmt-executor/main.go [hunk 1]
```diff
@@ -83,107 +82,85 @@ func main() {
...
+			LoggingClient.Error(fmt.Sprintf("docker command failed: %s", string(out)))
+			return err
 		}
 
+		// check that the command actually resulted in the correct state for
+		// the container
+		var expectedStatus bool
 		switch operation {
 		case START:
+			fallthrough
```
Comment (akramtexas) [inline cmd/sys-mgmt-executor/main.go]: I looked up the keyword `fallthrough` ([in the language reference](https://github.com/golang/go/wiki/Switch)) as I had not seen it being used _elsewhere_ in the `edgex-go` codebase... But this looks right to me, plus I found two instances of `fallthrough` (already) being used (both in `...internal/export/distro/registrations.go`), and which I had not noticed before 👍

**Comment by tsconn23** [comment]
After bringing up all services via the docker-compose file, I'm attempting to run `sys-mgmt-executor stop edgex-support-notifications` and receiving the following error. Container is not stopped. Am I doing something wrong?

```
~/Documents/code/git/tsconn23/edgex-go/cmd/sys-mgmt-executor $ ./sys-mgmt-executor stop edgex-support-notifications
level=INFO ts=2019-05-20T14:48:38.16675Z app=docker-compose-executor source=main.go:63 msg="This is the docker-compose-executor application!"
level=INFO ts=2019-05-20T14:48:38.167428Z app=docker-compose-executor source=main.go:66 msg="Application started in: 695.166µs"
level=ERROR ts=2019-05-20T14:48:38.167767Z app=docker-compose-executor source=main.go:78 msg="error performing  edgex-support-notifications on service stop: the service stop is an unknown service for which request was made to run Docker command"
```

**Comment by akramtexas** [comment]
> After bringing up all services via the docker-compose file, I'm attempting to run `sys-mgmt-executor stop edgex-support-notifications` and receiving the following error. Container is not stopped. Am I doing something wrong?
> 
> ```
> ~/Documents/code/git/tsconn23/edgex-go/cmd/sys-mgmt-executor $ ./sys-mgmt-executor stop edgex-support-notifications
> level=INFO ts=2019-05-20T14:48:38.16675Z app=docker-compose-executor source=main.go:63 msg="This is the docker-compose-executor application!"
> level=INFO ts=2019-05-20T14:48:38.167428Z app=docker-compose-executor source=main.go:66 msg="Application started in: 695.166µs"
> level=ERROR ts=2019-05-20T14:48:38.167767Z app=docker-compose-executor source=main.go:78 msg="error performing  edgex-support-notifications on service stop: the service stop is an unknown service for which request was made to run Docker command"
> ```

The command you are issuing looks right to me. I have been using a REST client for testing API calls and have _not_ seen any issue with running Docker commands ("start", "stop", or "restart")... Have you used `Portainer` as a tool to aid with getting insight into container logging? This is the relevant snippet from the `/edgexfoundry/developer-scripts/compose-files/docker-compose.yml` file:
```
  portainer:
    image:  portainer/portainer
    ports:
      - "9000:9000"
    command: -H unix:///var/run/docker.sock
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - portainer_data:/data
    depends_on:
      - volume  

networks:
  edgex-network:
    driver: "bridge"
```

**Comment by anonymouse64** [comment]
@tsconn23 my tests were written in the wrong order, it's supposed to be `sys-mgmt-executor <svc> <op>`. It works here:

```
$ make cmd/sys-mgmt-executor/sys-mgmt-executor 
CGO_ENABLED=0 GO111MODULE=on go build -ldflags "-X github.com/edgexfoundry/edgex-go.Version=1.0.0" -o cmd/sys-mgmt-executor/sys-mgmt-executor ./cmd/sys-mgmt-executor
$ ./cmd/sys-mgmt-executor/sys-mgmt-executor --help
Usage: ././cmd/sys-mgmt-executor/sys-mgmt-executor <service> <operation>                Start app with requested {service} and {operation}
       -h                                                       Show this message
$ ./cmd/sys-mgmt-executor/sys-mgmt-executor edgex-support-notifications stop
level=INFO ts=2019-05-20T15:50:46.021328218Z app=docker-compose-executor source=main.go:63 msg="This is the docker-compose-executor application!"
level=INFO ts=2019-05-20T15:50:46.025700226Z app=docker-compose-executor source=main.go:66 msg="Application started in: 4.399078ms"
level=INFO ts=2019-05-20T15:50:46.773420218Z app=docker-compose-executor source=main.go:116 msg="service container edgex-support-notifications is not running"
level=INFO ts=2019-05-20T15:50:46.777057519Z app=docker-compose-executor source=main.go:80 msg="success performing stop on service edgex-support-notifications"
ijohnson@ijohnson-XPS-15-9560:~/go/src/github.com/edgexfoundry/edgex-go$ 
```

**Comment by akramtexas** [comment]
As I mentioned, I've been using the REST client to invoke SMA. But on revisiting the source, it actually states that the order of arguments is like so:
`Usage: ./sys-mgmt-executor <service> <operation>`

**Pull request merged**